posthog/exception_capture.py

```diff
@@ -61,14 +62,15 @@ def exception_receiver(self, exc_info, extra_properties):
 
     def capture_exception(self, exception, metadata=None):
         try:
-            # if hasattr(sys, "ps1"):
-            #     # Disable the excepthook for interactive Python shells
-            #     return
+            distinct_id = metadata.get("distinct_id") if metadata else None
 
-            distinct_id = metadata.get("distinct_id") if metadata else DEFAULT_DISTINCT_ID
-            # Make sure we have a distinct_id if its empty in metadata
-            distinct_id = distinct_id or DEFAULT_DISTINCT_ID
+            # if there's no distinct_id, we'll generate one and set personless mode
+            # via $process_person_profile = false
+            properties = {}
+            if distinct_id is None:
+                properties["$process_person_profile"] = False
+                distinct_id = uuid.uuid4()
```

Comment:
Should we be using v7 going forward? Unrelated to this change

Reply:
Maybe? We don't really care what's in `distinct_id`, and don't make recommendations to customers about it I think - we probably should us v7 everywhere just for the sake of the habit, but the python standard library doesn't support it yet afaict.